core/src/main/java/org/apache/iceberg/DeleteFileIndex.java

```diff
@@ -462,7 +462,7 @@ DeleteFileIndex build() {
       Iterable<ManifestFile> matchingManifests = evalCache == null ? deleteManifests :
           Iterables.filter(deleteManifests, manifest ->
               manifest.content() == ManifestContent.DELETES &&
-                  (manifest.hasAddedFiles() || manifest.hasDeletedFiles()) &&
+                  (manifest.hasAddedFiles() || manifest.hasDeletedFiles() || manifest.hasExistingFiles()) &&
```

Comment:
There's no need for any of these checks now. Added, deleted, and existing are the only 3 options so this means all delete manifests.

Reply:
Actually, I'm wondering if the original intent of this was to check for added or existing files, but ignore manifests with only deleted files. I think it should probably be `(manifest.hasAddedFiles() || manifest.hasExistingFiles()) &&`. Can you try that out instead?

Reply:
I think it is correct. I'll try it.